### PR TITLE
feat(3ds): three DS SDK - adding logs to track milestone events

### PR DESCRIPTION
**Pull request by vsrivatsa-juspay**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Three DS SDK - adding logs to track milestone events

## Motivation and Context

Three DS authentication analytics and audit trail tracking

## How did you test it?

Check payment logs against a payment ID

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/ThreeDSMethod.res [hunk 1]
```diff
@@ -40,33 +40,63 @@ let make = () => {
           ->Belt.Option.flatMap(JSON.Decode.object)
           ->Belt.Option.flatMap(x => x->Dict.get("three_ds_method_data"))
           ->Option.getOr(Dict.make()->JSON.Encode.object)
+        let paymentIntentId =
+          metaDataDict
+          ->Dict.get("paymentIntentId")
```
Comment (PritishBudhiraja) [inline src/ThreeDSMethod.res:45]: getString function can be used I guess. Check once for both the values

File: src/ThreeDSAuth.res [hunk 2]
```diff
@@ -5,7 +5,7 @@ let make = () => {
   let (openModal, setOpenModal) = React.useState(_ => false)
   let (loader, setloader) = React.useState(_ => true)
 
-  let logger = Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom)
+  let logger = OrcaLogger.make()
```
Comment (ArushKapoorJuspay) [inline src/ThreeDSAuth.res:8]: Why are we creating a new logger instead of using the same from Recoil?

File: src/orca-loader/Elements.res [hunk 3]
```diff
@@ -121,7 +121,12 @@ let make = (
         let msg = [("paymentMethodList", json)]->Dict.fromArray
         mountedIframeRef->Window.iframePostMessage(msg)
         let maskedPayload = json->PaymentHelpers.maskPayload->JSON.stringify
-        logger.setLogInfo(~value=maskedPayload, ~eventName=PAYMENT_METHODS_RESPONSE, ())
+        logger.setLogInfo(
+          ~value="",
```
Comment (ArushKapoorJuspay) [inline src/orca-loader/Elements.res:125]: What is the difference between passing masked payload in value and in internalMetadata?

File: src/ThreeDSAuth.res (hunk 2 above)
Comment (vsrivatsa-juspay) [reply src/ThreeDSAuth.res:8]: This is a separate iframe.
Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom) returns a dummy instance of iframe

**Review by ArushKapoorJuspay — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Review by prafulkoppalkar — APPROVED** [review]

**Review by akash-c-k — APPROVED** [review]

**Pull request merged**